blockly-core/blocks/functional_procedures.js

```diff
@@ -224,6 +224,19 @@ Blockly.Blocks.functional_definition = {
       this.updateParams_();
     }
   },
+  changeParamType: function(name, newType) {
+    var change = false;
+    for (var x = 0; x < this.parameterNames_.length; x++) {
+      if (Blockly.Names.equals(name, this.parameterNames_[x])) {
+        this.parameterTypes_[x] = newType;
+        change = true;
+      }
+    }
```

Comment:
Right now, that's correct. I think it'd make sense to change them by ID instead, but IIRC it'll take a hefty change on the block space operation side to disambiguate for blocks already on the main blockspace